extensions/2.0/Khronos/KHR_materials_specular/schema/glTF.KHR_materials_specular.schema.json

```diff
@@ -0,0 +1,41 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema",
+    "title": "KHR_materials_specular glTF extension",
+    "type": "object",
+    "description": "glTF extension that defines the strength of the specular reflection.",
+    "allOf": [ { "$ref": "glTFProperty.schema.json" } ],
+    "properties": {
+        "specularFactor": {
+            "type": "number",
+            "description": "The strength of the specular reflection.",
+            "default": 1.0,
+            "minimum": 0.0,
+            "maximum": 1.0,
+            "gltf_detailedDescription": "This parameter scales the amount of specular reflection on non-metallic surfaces. It has no effect on metals."
+        },
+        "specularTexture": {
+            "allOf": [ { "$ref": "textureInfo.schema.json" } ],
+            "description": "A texture that defines the specular factor in the alpha channel.",
+            "gltf_detailedDescription": "A texture that defines the specular factor in the alpha channel. This will be multiplied by specularFactor."
+        },
+        "specularColorFactor": {
+            "type": "array",
+            "items": {
+                "type": "number",
+                "minimum": 0.0
```

Comment:
Why is there no `maximum` value here?

Reply:
Values above 1.0 are now clamped in a different part of the formula:

https://github.com/DassaultSystemes-Technology/glTF/blob/KHR_materials_specular/extensions/2.0/Khronos/KHR_materials_specular/README.md#implementation

This allows the specularColor to be above or below the IOR setting, while preserving energy conservation by clamping to 1.0 further down the line.

Reply:
For the non-IOR case this means that the effective maximum value for `specularColorFactor` is `[82365, 82365, 82365]` as it results in `dielectricSpecularF0` being equal to `1.0` for all non-zero texture channels, assuming 8-bit sRGB encoding. Is it a valid use case?

The regular glTF 2.0 convention is that `Factor * Texture` is "pre-clamped" by design. Deferring such clamping to the formula has an interesting implication: since the _Implementation_ section is marked as non-normative, the location of the clamping step (i.e. after multiplication with `0.04` or IOR parameter) and even the presence of clamping are no longer ratified so the `specularColor` parameter becomes even more undefined.

Reply:
It's not common to specify a `dielectricSpecularF0` that high, as typically that would suggest something metallic, but we didn't want to forbid it here.

Slightly off-topic, but we're also introducing an unclamped `thicknessFactor` in KHR_materials_volume, which represents a distance in mesh space, typically the longest edge of a given mesh's bounding box.  In the corresponding texture, white indicates an average thickness of the full width of that thicknessFactor, and darker colors indicate thinner parts.

Reply:
> we didn't want to forbid it here

OK, should the spec be more prescriptive about clamping behavior, i.e. by additionally defining it in the normative section?

Since thickness is defined in meters, values greater than `1.0` are perfectly fine there.

Reply:
The clamping behavior is mentioned in the last paragraph of the normative part:

> The specular color factor is allowed to be set to color values greater than [1, 1, 1]. Thus, the reflection amount can go beyond what is determined by the index of refraction (IOR). To still ensure energy conservation, the product of specular color and f0 reflectance from IOR is clamped. Please refer to Implementation for detailed instructions on where to place the clamping operation.

Reply:
That paragraph is exactly what concerns me here.
- > The specular color factor is allowed to be set to color values greater than [1, 1, 1]

  We shouldn't call an intermediate product of the factor and the texture value a "color", since that term is loosely defined for values above 1.

- > Please refer to Implementation ... on where to place the clamping operation.

  A normative section may refer to a non-normative section only for providing an example, not for detailed definitions.

I suggest rephrasing it as (self-contained but slightly more verbose):
> The specular color factor is allowed to be set to values greater than [1, 1, 1]. Thus, the reflection amount can go beyond what is determined by the index of refraction (IOR). To still ensure energy conservation, the product of specular color factor, specular color texture, and f0 reflectance from IOR is clamped to `1.0`. Please refer to Implementation for an example on where to place the clamping operation.

Reply:
OK, I understand the concern. Thanks for the suggestion, I like it. Maybe let's wait for a quick thumbs up from @emackey and then merge it.

Reply:
That sounds good to me too, thanks.

Reply:
Done, thanks!